### PR TITLE
Resolving subcommand and subcommand group for CommandInteractions

### PR DESCRIPTION
**Pull request by jaydamani**
**Please describe the changes this PR makes and why it should be merged:**

Added CommandInteraction#subcommand and CommandInteraction#group

**Status and versioning classification:**

- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Comment by monbrey** [comment]
I don't see how this is necessary - there are no issues with sub command or sub command group options.

Why do you believe this change is needed?

**Comment by jaydamani** [comment]
With this, lib would do the parsing and figure out the subcommand / subcommand group while options is just a map of arguments provided by the user. Not necessary but could be kinda handy.

**Comment by monbrey** [comment]
It's not "parsing" anything though. Looks to me like you just hoist the options up a level, but then don't do any of the existing handling on them.

Just seems worse to me.

**Comment by monbrey** [comment]
@jaydamani Hey, apologies, I think I completely misread the diff on this change. I've just tested your proposed changes and I think this could be beneficial. Do you want to reopen this for reviews?

**Comment by jaydamani** [comment]
No problem, sorry for the delay ~~I should check my mail more often~~, kinda my bad as well, since I couldn't explain it properly

File: src/structures/CommandInteraction.js [hunk 1]
```diff
@@ -128,6 +126,22 @@ class CommandInteraction extends Interaction {
    */
   _createOptionsCollection(options, resolved) {
     const optionsCollection = new Collection();
+
+    if (options?.[0].type === 2) {
+      /**
+       * @type {string}
+       */
+      this.group = options[0].name;
```
Comment (NotSugden) [inline src/structures/CommandInteraction.js:134]: isn"t there an enum for this?

File: src/structures/CommandInteraction.js [hunk 2]
```diff
@@ -128,6 +126,22 @@ class CommandInteraction extends Interaction {
...
+      options = options[0].options;
+    }
+    if (options?.[0].type === 1) {
+      /**
+       * @type {string}
```
Comment (NotSugden) [inline src/structures/CommandInteraction.js:139]: ...? description?

File: src/structures/CommandInteraction.js [hunk 3]
```diff
@@ -128,6 +126,22 @@ class CommandInteraction extends Interaction {
    */
   _createOptionsCollection(options, resolved) {
     const optionsCollection = new Collection();
+
+    if (options?.[0].type === 2) {
+      /**
+       * @type {string}
```
Comment (NotSugden) [inline src/structures/CommandInteraction.js:132]: ^

**Review by NotSugden — CHANGES_REQUESTED** [review]

**Comment by iCrawl** [comment]
This will be handled in #6107

**Pull request closed without merging**